uci.go

```diff
@@ -127,6 +131,22 @@ func (t *tree) Revert(configs ...string) {
 	t.Unlock()
 }
 
+func (t *tree) GetSections(config string, secType string) ([]string, bool) {
```

Comment:
The parent commit didn't have the `secType` parameter, and returned all sections.

Should `secType == ""` still be a special case, or do we want a `GetAllSections(config string)` method? If that is not needed at all, please squash your commits.

Reply:
(Sorry I didn't get to reply earlier, things got a little bit chaotic here.)

SecType == "" is in my opinion not a really a useful case.
When I first implemented it, I somehow overlooked that the same file can have different sections in it.  Or at least it didn't matter for that specific use case.
However it didn't took long until that stopped working for me.

In my opinion it is nicer/cleaner (and more intuitive) to have the capability to filter by type,
than having to iterate over all section names, get the section and then determine their type.

Reply:
Fair point.